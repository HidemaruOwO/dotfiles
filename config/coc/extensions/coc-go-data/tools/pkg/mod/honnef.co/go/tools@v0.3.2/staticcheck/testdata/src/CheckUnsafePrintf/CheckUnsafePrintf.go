package pkg

import (
	"fmt"
	"log"
	"os"
)

func fn(s string) {
	fn2 := func() string { return "" }
	fmt.Printf(fn2())      //@ diag(`should use print-style function`)
	_ = fmt.Sprintf(fn2()) //@ diag(`should use print-style function`)
	log.Printf(fn2())      //@ diag(`should use print-style function`)
	fmt.Printf(s)          //@ diag(`should use print-style function`)
	fmt.Printf(s, "")
	fmt.Fprintf(os.Stdout, s) //@ diag(`should use print-style function`)
	fmt.Fprintf(os.Stdout, s, "")

	fmt.Printf(fn2(), "")
	fmt.Printf("")
	fmt.Printf("%s", "")
	fmt.Printf(fn3())
}

func fn3() (string, int) { return "", 0 }
