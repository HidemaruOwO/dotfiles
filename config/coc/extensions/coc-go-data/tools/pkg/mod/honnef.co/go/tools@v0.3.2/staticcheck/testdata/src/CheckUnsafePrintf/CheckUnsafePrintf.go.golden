package pkg

import (
	"fmt"
	"log"
	"os"
)

func fn(s string) {
	fn2 := func() string { return "" }
	fmt.Print(fn2())      //@ diag(`should use print-style function`)
	_ = fmt.Sprint(fn2()) //@ diag(`should use print-style function`)
	log.Print(fn2())      //@ diag(`should use print-style function`)
	fmt.Print(s)          //@ diag(`should use print-style function`)
	fmt.Printf(s, "")
	fmt.Fprint(os.Stdout, s) //@ diag(`should use print-style function`)
	fmt.Fprintf(os.Stdout, s, "")

	fmt.Printf(fn2(), "")
	fmt.Printf("")
	fmt.Printf("%s", "")
	fmt.Printf(fn3())
}

func fn3() (string, int) { return "", 0 }
